### ISSUE TITLE
Typescript optional parameter/field syntax (?:)

### ISSUE THREAD

**Issue by mst101** [shikijs/shiki#387]
When specifying an optional parameter or field within a typescript fenced code block, both the question mark and colon are omitted: 

```ts
function test(optionalParam?: string) {
  return `test${optionalParam}`
}

interface Test {
  optionalField?: string
}
```

[See example here](https://nuxt.com/docs/guide/directory-structure/pages#typing-custom-metadata)

**Comment by muenzpraeger** [comment]
Also fixed with #377

**Comment by muenzpraeger** [comment]
[0.12.0 with the aforementioned PR](https://github.com/shikijs/shiki/releases/tag/v0.12.0) is out.
